pkg/dns/treecache.go

```diff
@@ -59,61 +50,33 @@ func (cache *TreeCache) Serialize() (string, error) {
 }
 
 func (cache *TreeCache) setEntry(key string, val interface{}, path ...string) {
-	if cache.rwLock != nil {
-		cache.rwLock.Lock()
-		defer cache.rwLock.Unlock()
-	}
 	node := cache.ensureChildNode(path...)
 	node.Entries[key] = val
 }
 
-func (cache *TreeCache) replaceEntries(entries map[string]interface{}, path ...string) {
-	if cache.rwLock != nil {
-		cache.rwLock.Lock()
-		defer cache.rwLock.Unlock()
-	}
-	node := cache.ensureChildNode(path...)
-	node.Entries = make(map[string]interface{})
-	for key, val := range entries {
-		node.Entries[key] = val
-	}
-}
-
 func (cache *TreeCache) getSubCache(path ...string) *TreeCache {
 	childCache := cache
 	for _, subpath := range path {
 		childCache = childCache.ChildNodes[subpath]
 		if childCache == nil {
-			return childCache
+			return nil
 		}
 	}
 	return childCache
 }
 
 func (cache *TreeCache) setSubCache(key string, subCache *TreeCache, path ...string) {
```

Comment:
I don't think the comment would make sense here. the treecache is not thread-safe.
It could be used by the same thread/go-routine without locking,
or by multiple threads, using a simple mutex or rwmutex. 
it is up to the caller to choose..